crates/sui-types/src/passkey_authenticator.rs

```diff
@@ -0,0 +1,207 @@
+    #[serde(skip)]
+    parsed_challenge: Vec<u8>,
+    #[serde(skip)]
+    bytes: OnceCell<Vec<u8>>,
+}
+
+impl PasskeyAuthenticator {
+    /// Validate client_data_json and initialize parsed challenge.
+    pub fn init(&mut self) -> Result<Self, SuiError> {
+        let client_data_json: CollectedClientData = serde_json::from_slice(&self.client_data_json)
+            .map_err(|_| SuiError::InvalidSignature {
+                error: "Failed to parse client data".to_string(),
+            })?;
+        if client_data_json.ty != ClientDataType::Get {
+            return Err(SuiError::InvalidSignature {
+                error: "Invalid client data type".to_string(),
+            });
+        }
+        let parsed_challenge =
+            Base64UrlUnpadded::decode_vec(&client_data_json.challenge).map_err(|e| {
+                SuiError::InvalidSignature {
+                    error: e.to_string(),
+                }
+            })?;
+        self.parsed_challenge = parsed_challenge;
+        Ok(self.to_owned())
+    }
+
+    pub fn new_for_testing(
+        authenticator_data: Vec<u8>,
+        client_data_json: Vec<u8>,
+        user_signature: Signature,
+    ) -> Result<Self, SuiError> {
+        let mut passkey = PasskeyAuthenticator {
+            authenticator_data,
+            client_data_json,
+            user_signature,
+            parsed_challenge: Default::default(),
+            bytes: OnceCell::new(),
+        };
+        passkey.init()
+    }
+
+    pub fn get_pk(&self) -> SuiResult<PublicKey> {
+        PublicKey::try_from_bytes(
+            SignatureScheme::Secp256r1,
+            self.user_signature.public_key_bytes(),
+        )
+        .map_err(|_| SuiError::InvalidAuthenticator)
+    }
+}
+
+/// Necessary trait for [struct SenderSignedData].
+impl PartialEq for PasskeyAuthenticator {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_ref() == other.as_ref()
+    }
+}
+
+/// Necessary trait for [struct SenderSignedData].
+impl Eq for PasskeyAuthenticator {}
+
+/// Necessary trait for [struct SenderSignedData].
+impl Hash for PasskeyAuthenticator {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.as_ref().hash(state);
+    }
+}
+
+impl AuthenticatorTrait for PasskeyAuthenticator {
+    fn verify_user_authenticator_epoch(
+        &self,
+        _epoch: EpochId,
+        _max_epoch_upper_bound_delta: Option<u64>,
+    ) -> SuiResult {
+        Ok(())
+    }
+
+    /// Verify an intent message of a transaction with an passkey authenticator.
+    fn verify_claims<T>(
+        &self,
+        intent_msg: &IntentMessage<T>,
+        author: SuiAddress,
+        _aux_verify_data: &VerifyParams,
+        _zklogin_inputs_cache: Arc<VerifiedDigestCache<ZKLoginInputsDigest>>,
+    ) -> SuiResult
+    where
+        T: Serialize,
+    {
+        let mut hasher = DefaultHash::default();
+        hasher.update(&bcs::to_bytes(&intent_msg).expect("Message serialization should not fail"));
```

Comment:
please use 

```
        bcs::serialize_into(&mut hasher, &intent_msg)
            .expect("Message serialization should not fail");
```

to avoid allocating unnecessarily